docs/development/build-instructions-gn.md

```diff
@@ -0,0 +1,93 @@
+# Build Instructions (experimental GN build)
+
+Follow the guidelines below for building Electron with the experimental GN
+build.
+
+> **NOTE**: The GN build system is in _experimental_ status, and currently only
+> works on macOS and Linux, in debug mode, as a component build.
+
+## Prerequisites
+
+See the [macOS](build-instructions-osx.md#prerequisites) or
+[Linux](build-instructions-linux.md#prerequisites) build instructions for the
+requirements for your platform. In addition, you'll need to install
+[`depot_tools`][depot-tools], the toolset used for fetching Chromium and its
+dependencies.
+
+[depot-tools]: http://commondatastorage.googleapis.com/chrome-infra-docs/flat/depot_tools/docs/html/depot_tools_tutorial.html#_setting_up
+
+## Getting the Code
+
+```sh
+$ mkdir electron-gn && cd electron-gn
+$ cat > .gclient <<-GCLIENT
+solutions = [
+  {
+    "url": "https://github.com/electron/electron",
+    "managed": False,
+    "name": "src/electron",
+  },
+]
+GCLIENT
+$ gclient sync --with_branch_heads --with_tags
+# This will take a while, go get a coffee.
+```
+
+## Building
+
+```sh
+$ cd src
+$ export CHROMIUM_BUILDTOOLS_PATH=`pwd`/buildtools
+$ gn gen out/Default --args='root_extra_deps=["//electron"] is_electron_build=true is_component_build=true use_jumbo_build=true'
```

Comment:
Can you also add the other required flags from https://github.com/electron/libchromiumcontent/blob/master/chromiumcontent/args/shared_library.gn for example `v8_*` which is required by node. We seem to checkout nacl, in that case should we add `enable_nacl=false` ?

Reply:
chrome 63 doesn't have a way to disable checking out nacl. As far as I can tell, `enable_nacl` doesn't actually affect anything we depend on (I grepped and looked at all the places it's referenced, and also I don't have any nacl-related build artifacts in my `out/Default` directory, despite the fact that I haven't set `enable_nacl = false`, so I conclude that it's not necessary). I'll add the `v8_*`, though it seems like v8's been working OK without them... maybe it's subtly broken though and I just haven't noticed!

I'm also looking into better ways to specify these args than passing them all in the command line copy/pasted from the docs.